src/ResourceManager.cpp

```diff
@@ -0,0 +1,46 @@
+#include "ResourceManager.h"
+
+ResourceManager::ResourceManager(){}
+ResourceManager::~ResourceManager(){}
+
+void ResourceManager::LoadResources() {
+	LoadTexture("Duck", "Res/JanitorDuck.png");
+	LoadTexture("Meadow", "Res/quackMeadowBG.png");
+
+	LoadSound("Quack", "Res/quack.wav");
+	LoadSound("Frog", "Res/froggy.wav");
+	LoadSound("Catch", "Res/frogCatch.wav");
+
+	LoadFont("Text", "Res/consola.ttf");
+}
+
+void ResourceManager::LoadTexture(const char* name, const char* path) {
+    sf::Texture& tex = TextureMap[name]; 
+	tex.loadFromFile(path);
+};
+
+void ResourceManager::LoadSound(const char* name, const char* path) {
+	sf::SoundBuffer& Buffer = BufferMap[name];
+	Buffer.loadFromFile(path);
+	
+	sf::Sound& Sound = SoundMap[name];
+	Sound.setBuffer(Buffer);
+};
+
+void ResourceManager::LoadFont(const char* name, const char* path) {
+	sf::Font& font = FontMap[name];
+	font.loadFromFile(path);
+};
+
+
+sf::Texture& ResourceManager::GetTexture(const char* name) {
+	return TextureMap.at(name);
```

Comment:
FYI - getting a resource that doesn't exist will crash the game - at() throws an exception when it doesn't find the entry in the map.

It's good that it crashes, just adding this comment for educational purpose :D